### ISSUE TITLE
question about details of parameters

### ISSUE THREAD

**Issue by 1245244103** [HKUNLP/icl-ceil#5]
hello, i am  trying to reproduce the result in the paper.I run the scripts/run_epr.sh successfully. i get the em score of about 71 in mrpc which is 75.98 in the paper.Are the settings in run_epr.sh different from the paper? Can you provide the setting parameters in the paper?
Thanks a lot!

**Comment by jiacheng-ye** [comment]
Hi, it's weird as the settings in run_epr.sh is the same as that in the paper. Could you check whether you can obtain similar results to the paper for other methods such as Topk-BERT as I'm not sure if it's due to the randomness of running on different machines.

**Comment by hanxinyan20** [comment]
I found the same question, too. I run the script/run_bm25.sh, and got the acc on mrpc validation set is 0.576 which is far lower than in the paper. I just change the num_ice to 27, and leave other parameters unchanged. I also tried to evaluate sst5 ( set num_ice to 27), the acc is 0.296. Can you give me some advice so that I can get the same acc as you claimed in your paper?

**Comment by 1245244103** [comment]
> I found the same question, too. I run the script/run_bm25.sh, and got the acc on mrpc validation set is 0.576 which is far lower than in the paper. I just change the num_ice to 27, and leave other parameters unchanged. I also tried to evaluate sst5 ( set num_ice to 27), the acc is 0.296. Can you give me some advice so that I can get the same acc as you claimed in your paper?

While replicating the process, I noticed issues with the training code for the encoder. I've rewritten the training code without using a trainer, following the style of Hugging Face, and without the use of accelerate. Additionally, there were some discrepancies between the parameters used in the code and those described in the paper. For instance, the paper mentions using the three samples with the highest and lowest scores as positive and negative examples, respectively, whereas the code only samples one. I have made adjustments to align with the paper. After these modifications, the results on some datasets are close to those reported in the paper. You might want to give it a try.
